### ISSUE TITLE
Block Editor: Unable to Edit Text with Screen Reader

### ISSUE THREAD

**Issue by donalirl** [wordpress-mobile/WordPress-iOS#14007]
### Context
I am opening this up on behalf of a visually-impaired user who used Apple Voiceover, but could use some help documenting the issue as I don't have an iPhone.

They report that the paragraph blocks have buttons that should allow them to edit text, but it doesn't work. They have to go back up to a more options at the top, double tap on it, then scroll all the way down to the bottom where it says 'Continue editing', and then it allows to edit the the text in the block.

### Expected behavior
They expect to be able to edit the text as soon as they tap on the paragraph block.

### Steps to reproduce the behavior
[needed]

I opened a corresponding issue in the Android app repo WordPress-Android/issues/11765 but based on this user's description, I feel the experience is very different.
